### PR TITLE
Do not process virtual modules from other plugins

### PR DESCRIPTION
**Pull request by proteriax**
Fixes proteriax/rollup-plugin-ignore#4.

## Rollup Plugin Name: `{commonjs}`

This PR contains:

- [x] bugfix
- [ ] feature
- [ ] refactor
- [ ] documentation
- [ ] other

Are tests included?

- [ ] yes (_bugfixes and features will not be merged without tests_)
- [x] no

Breaking Changes?

- [ ] yes (_breaking changes will not be merged unless absolutely necessary_)
- [x] no

If yes, then include "BREAKING CHANGES:" in the first commit message body, followed by a description of what is breaking. 

List any relevant issue numbers:
proteriax/rollup-plugin-ignore#4
rollup/rollup-plugin-commonjs#315
rollup/plugins#248

### Description

According to rollup’s documentation,
> If your plugin uses 'virtual modules' (e.g. for helper functions), prefix the module ID with \0. This prevents other plugins from trying to process it.

The current behavior of `@rollup/plugin-commonjs` violates this presumption by processing a module starting with `\0` included by `require`.

Suppose the following module:
```js
const fs = require('fs')
console.log(fs)
```

It will be overwritten to `\0rollup_plugin_ignore_empty_module_placeholder?commonjs-proxy`, but because `@rollup/plugin-commonjs` provide the contents to all modules with its own suffix without checking `\0`, `rollup-plugin-ignore` would not have a chance to supply the content for this virtual module.

### CODE REVIEW

**Comment by TrySound** [comment]
Could you add test case so nobody break the logic again?

File: packages/commonjs/src/resolve-id.js [hunk 1]
```diff
@@ -45,17 +45,20 @@ export default function getResolveId(extensions) {
...
         importee.startsWith(DYNAMIC_JSON_PREFIX)
       ) {
         return importee;
+      } else if (isProxyModule) {
+        importee = getIdFromProxyId(importee);
+      } else {
+        return null;
       }
-      return null;
+    } else if (isProxyModule) {
```
Comment (lukastaegert) [inline packages/commonjs/src/resolve-id.js:60]: A proxy module always starts with `\0` so it is impossible to hit that branch.

**Comment by shellscape** [comment]
@proteriax are you interested in adding tests to this PR so we can approve it?

**Comment by proteriax** [comment]
@shellscape I’ll close it for now since I don’t have time for this issue at this moment.

**Pull request closed without merging**